### PR TITLE
#274 Fix formatting content in composer

### PR DESCRIPTION
**Pull request by oubchid**
Demo: https://streamable.com/nyithf

### CODE REVIEW

**Comment by MichaelBailly** [comment]
I do not agree with this solution. You prevent any HTML to be pasted in the composer, everything will be converted to plain text. This is a regression in terms of features.

**Comment by oubchid** [comment]
> I do not agree with this solution. You prevent any HTML to be pasted in the composer, everything will be converted to plain text. This is a regression in terms of features.

So we need to keep `some allowed` html tags

**Comment by oubchid** [comment]
Ok so my analysis, the problem comes from a paragraph with a style properties `line-height` defined to `1.15px`. Removing this the problem is fixed!
But the question, Why we have two libs in the project, and why using `esnDomPurify` instead of `sanitizeStylisedHtml` [here](https://github.com/OpenPaaS-Suite/esn-frontend-inbox/blob/83131665e0dcfa068b5da3f5b8196ed65805bdb4/src/esn.inbox.libs/app/components/message-body/html/message-body-html.pug#L4) ? 


Tip: 
- using `sanitizeStylisedHtml` filter fixes the problem.
- DOMPurify will NOT prevent you from XSS caused by crazy library features as for example AngularJS has. [explained here](https://github.com/cure53/DOMPurify/wiki/Security-Goals-&-Threat-Model)

File: src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.spec.js [hunk 1]
```diff
@@ -146,6 +153,31 @@ describe('The inboxComposerBodyEditorHtml component', function() {
...
+          getData: () => unsafeHtml
+        }
+      };
+
+      controller.onSummernotePaste(onPasteEvent);
+      $rootScope.$digest();
+
+      const trustedValue = $filter('sanitizeStylisedHtml')(unsafeHtml).$$unwrapTrustedValue();
+
+      expect(trustedValue).to.equal('<h1>hello</h1><img src="x" />');
```
Comment (rezk2ll) [inline src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.spec.js:172]: should be something like:
```js
expect(document.execCommand).to.have.been.calledWith('insertHtml', false, '<h1>hello</h1><img src="x" />');
```

Comment (oubchid) [reply src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.spec.js:172]: I have already tested this, but it does not work to mock `document`

**Review by alagane — DISMISSED** [review]

**Review by alagane — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**